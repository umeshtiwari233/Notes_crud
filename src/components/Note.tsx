import {
  TouchableOpacity,
  Text,
  View,
  StyleSheet,
  Dimensions,
  Alert,
  Pressable,
  Image,
} from "react-native";
import React, { Component } from "react";
import colors from "../utils/colors";
import AsyncStorage from "@react-native-async-storage/async-storage";
import NoteInputModal from "../modules/NoteInputModal";

type MyProps = {
  item: any;
  findNotes: any;
};
type MyState = {
  note:{title: string; desc: string };
  modalVisible:boolean
};

export class Note extends Component<MyProps, MyState> {
  constructor(props:MyProps){
    super(props)
    this.state = {
      modalVisible : false , 
      note : {title : this.props.item.title , desc : this.props.item.desc}
    }
  }

  closeModal = ()=> {
    this.setState({
        modalVisible : false
    })
  }

  openModal = ()=> {
    this.setState({
        modalVisible : true
    })
  }

  handleUpdate =async (title : string , desc : string , time:any)=> {
    this.setState({
      note : {title : title , desc : desc}
    })
    const result = await AsyncStorage.getItem('notes')
    let notes = []
    if(result!== null){
      notes = JSON.parse(result)
    }

    const newNotes =  notes.filter(n => {
      if(n.id === this.props.item.id){
        n.title = title
        n.desc = desc 
        n.isUpdated = true 
        n.time = time
      }
      return n; 
    })  

    await AsyncStorage.setItem('notes' , JSON.stringify(newNotes))
    this.props.findNotes();
  }


  deleteNote = async () => {
    const result = await AsyncStorage.getItem("notes");
    let notes = [];
    if (result !== null) {
      notes = JSON.parse(result);
    }

    const newNotes = notes.filter(
      (n) => n.id !== this.props.item.id
    );
    console.log("newNote", newNotes);
    await AsyncStorage.setItem("notes", JSON.stringify(newNotes));
    this.props.findNotes();
  };

  displayAlert = () => {
    Alert.alert(
      "Are you sure!",
      "This action delete this note permanently",
      [
        {
          text: "Delete",
          onPress: () => this.deleteNote(),
        },
        {
          text: "Cancel",
          onPress: () => console.log("cancel"),
        },
      ],
      {
        cancelable: true,
      }
    );
  };
  render() {
    const { title, desc } = this.props.item;
    return (
      <View>
       <View style={styles.container} >
        <View style={styles.iconsTitleContainer} >
          <View>
          <Text style={styles.title} numberOfLines={2} >{title}</Text>
          <Text style={styles.desc} numberOfLines={3} >{desc}</Text>
          </View>
          <View style={styles.icons} >
            <Pressable onPress={this.openModal}>
                <Image source={{uri:'https://cdn-icons-png.flaticon.com/128/2951/2951136.png'}} style={styles.icon1}/>
            </Pressable>
            <Pressable onPress={this.displayAlert}>
                <Image source={{uri:'https://cdn-icons-png.flaticon.com/128/10374/10374182.png'}} style={styles.icon}/>
            </Pressable>
          </View>
        </View>
        
      </View>
      
      <NoteInputModal onSubmit={this.handleUpdate} isEdit={true} onClose={this.closeModal} note={{title : this.state.note.title , desc : this.state.note.desc}} visible={this.state.modalVisible} />
     </View>
    );
  }
}

const width = Dimensions.get("window").width - 40;

const styles = StyleSheet.create({
  container: {
    backgroundColor:'white',
    width: width - 10,
    height:100,
    padding: 15,
    marginLeft:5,
    marginVertical:8,
    borderRadius: 15,
  },
  title: {
    fontWeight: "bold",
    fontSize: 16,
    color: 'black',
  },
  iconsTitleContainer : {
    display   : 'flex' , 
    flexDirection : 'row' ,
    justifyContent : 'space-between'
  } , 
  icons : {
    width : 50,
  } , 
  icon : {
    width:30,
    height:30,
    
  },
  icon1 : {
    width:25,
    height:25,
    marginLeft:3,
    marginBottom:15
    
  },
  desc: {paddingTop:8},
});

export default Note;
