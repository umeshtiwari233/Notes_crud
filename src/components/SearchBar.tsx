import { Text, View } from "react-native";
import React, { Component } from "react";
import { StyleSheet } from "react-native";
import colors from "../utils/colors";
import { TextInput } from "react-native";

type MyProps = {};
type MyState = {};

export class SearchBar extends Component<MyProps, MyState> {
  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.searchBar} placeholder="Search here ..." />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginVertical: 20,
  },
  searchBar: {
    borderWidth: 0.5,
    borderColor: colors.PRIMARY,
    height: 40,
    borderRadius: 40,
    paddingLeft: 15,
  },
});

export default SearchBar;
