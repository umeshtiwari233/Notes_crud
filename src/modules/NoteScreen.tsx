import {
  Keyboard,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
  FlatList,
  Pressable,
  Image
} from "react-native";
import React, { Component } from "react";
import colors from "../utils/colors";
import SearchBar from "../components/SearchBar";
import NoteInputModal from "./NoteInputModal";
import AsyncStorage from "@react-native-async-storage/async-storage";
import Note from "../components/Note";
import Strings from "../utils/Strings";

type MyProps = {};

type MyState = {
  modalVisible: boolean;
  notes: { id: string; title: string; desc: string }[];
};

export class NoteScreen extends Component<MyProps, MyState> {
  constructor(props: MyProps) {
    super(props);
    this.state = {
      modalVisible: false,
      notes: [],
    };
  }

  componentDidMount(): void {
    this.findNotes();
  }

  findNotes = async () => {
    const result = await AsyncStorage.getItem("notes");
    console.log(result, "result");
    if (result !== null) {
      this.setState({
        notes: JSON.parse(result),
      });
    }
  };

  handleSubmit = async (title: string, desc: string) => {
    const note = { id: Date.now().toString(), title, desc, time: Date.now() };
    const updatedNotes = [...this.state.notes, note];
    this.setState({
      notes: updatedNotes,
    });
    await AsyncStorage.setItem("notes", JSON.stringify(updatedNotes));
  };

  openModal = () => {
    this.setState({
      modalVisible: true,
    });
  };

  closeModal = () => {
    this.setState({
      modalVisible: false,
    });
  };

  render() {
    return (
      <>
        <StatusBar translucent backgroundColor={'transparent'} />
        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
          <View style={styles.container}>
            <Text style={styles.header}>{Strings.Head_greet}</Text>
            <SearchBar />

            <FlatList
              data={this.state.notes}
              keyExtractor={(item) => item.id.toString()}
              showsVerticalScrollIndicator={false}
              renderItem={({item}) => <Note findNotes={this.findNotes} item={item} /> }
            />
            {!this.state.notes.length ? (
              <View style={styles.emptyHeadingContainer}>
                <Text style={styles.emptyHeader}>
                  {Strings.Background_text}
                </Text>
              </View>
            ) : null}
            <Pressable onPress={this.openModal}>
                <Image source={{uri:'https://cdn-icons-png.flaticon.com/128/10695/10695021.png'}} style={styles.touchableopacity}/>
            </Pressable>
           
          </View>
        </TouchableWithoutFeedback>

        <NoteInputModal
          isEdit={false}
          note={{ title: "", desc: "" }}
          visible={this.state.modalVisible}
          onClose={this.closeModal}
          onSubmit={this.handleSubmit}
        />
      </>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop:40,
    paddingHorizontal: 20,
    flex: 1,
    zIndex: 1,
  },
  header: {
    fontSize: 25,
    color:'black',
    fontWeight: "bold",
  },
  emptyHeadingContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    zIndex: -1,
  },
  emptyHeader: {
    fontSize: 30,
    textTransform: "uppercase",
    fontWeight: "bold",
  },
  touchableopacity: {
    width: 60,
    height: 60,
    borderRadius: 40,
    position: "absolute",
    right: 10,
    bottom: 20,
  },
  add: {
    fontSize: 30,
    color: "white",
  },
});

export default NoteScreen;
