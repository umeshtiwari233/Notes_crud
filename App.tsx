import React, { Component } from "react";
import NoteScreen from "./src/modules/NoteScreen";

export class App extends Component {
  render() {
    return (
      <NoteScreen/>
    );
  }
}

export default App;
