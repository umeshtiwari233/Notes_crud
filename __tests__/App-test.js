import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import SearchBar from '../src/components/SearchBar';
import NoteScreen from '../src/modules/NoteScreen';
import Note from '../src/components/Note';

test('snapshot testing',()=>{
  let Notescreenobj=renderer.create(<NoteScreen/>).getInstance();
  // let Homeobj = renderer.create(<Note findNotes={Notescreenobj.findNotes} item={Notescreenobj.state.notes} />).getInstance();
    
  const snap=renderer.create(
    <App/>
  ).toJSON();
  const snap1=renderer.create(
    <SearchBar/>
  ).toJSON();
  const snap2=renderer.create(
    <NoteScreen/>
  ).toJSON();
  const snap3=renderer.create(
    <App/>
  ).toJSON();
  const snap4=renderer.create(
    <Note findNotes={Notescreenobj.findNotes} item={Notescreenobj.state.notes} />
  ).toJSON();
  expect(snap).toMatchSnapshot();
  expect(snap1).toMatchSnapshot();
  expect(snap2).toMatchSnapshot();
  expect(snap3).toMatchSnapshot();
  expect(snap4).toMatchSnapshot();
})