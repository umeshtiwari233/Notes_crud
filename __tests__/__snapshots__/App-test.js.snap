// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flex": 1,
        "marginTop": 40,
        "paddingHorizontal": 20,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Hello Umesh
    </Text>
    <View
      style={
        {
          "marginVertical": 20,
        }
      }
    >
      <TextInput
        placeholder="Search here ..."
        style={
          {
            "borderColor": "black",
            "borderRadius": 40,
            "borderWidth": 0.5,
            "height": 40,
            "paddingLeft": 15,
          }
        }
      />
    </View>
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "zIndex": -1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 30,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        ADD NOTES
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={
          {
            "uri": "https://cdn-icons-png.flaticon.com/128/10695/10695021.png",
          }
        }
        style={
          {
            "borderRadius": 40,
            "bottom": 20,
            "height": 60,
            "position": "absolute",
            "right": 10,
            "width": 60,
          }
        }
      />
    </View>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  />,
]
`;

exports[`snapshot testing 2`] = `
<View
  style={
    {
      "marginVertical": 20,
    }
  }
>
  <TextInput
    placeholder="Search here ..."
    style={
      {
        "borderColor": "black",
        "borderRadius": 40,
        "borderWidth": 0.5,
        "height": 40,
        "paddingLeft": 15,
      }
    }
  />
</View>
`;

exports[`snapshot testing 3`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flex": 1,
        "marginTop": 40,
        "paddingHorizontal": 20,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Hello Umesh
    </Text>
    <View
      style={
        {
          "marginVertical": 20,
        }
      }
    >
      <TextInput
        placeholder="Search here ..."
        style={
          {
            "borderColor": "black",
            "borderRadius": 40,
            "borderWidth": 0.5,
            "height": 40,
            "paddingLeft": 15,
          }
        }
      />
    </View>
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "zIndex": -1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 30,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        ADD NOTES
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={
          {
            "uri": "https://cdn-icons-png.flaticon.com/128/10695/10695021.png",
          }
        }
        style={
          {
            "borderRadius": 40,
            "bottom": 20,
            "height": 60,
            "position": "absolute",
            "right": 10,
            "width": 60,
          }
        }
      />
    </View>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  />,
]
`;

exports[`snapshot testing 4`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flex": 1,
        "marginTop": 40,
        "paddingHorizontal": 20,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Hello Umesh
    </Text>
    <View
      style={
        {
          "marginVertical": 20,
        }
      }
    >
      <TextInput
        placeholder="Search here ..."
        style={
          {
            "borderColor": "black",
            "borderRadius": 40,
            "borderWidth": 0.5,
            "height": 40,
            "paddingLeft": 15,
          }
        }
      />
    </View>
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "zIndex": -1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 30,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        ADD NOTES
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={
          {
            "uri": "https://cdn-icons-png.flaticon.com/128/10695/10695021.png",
          }
        }
        style={
          {
            "borderRadius": 40,
            "bottom": 20,
            "height": 60,
            "position": "absolute",
            "right": 10,
            "width": 60,
          }
        }
      />
    </View>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  />,
]
`;

exports[`snapshot testing 5`] = `
<View
  testID="Note"
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 100,
        "marginLeft": 5,
        "marginVertical": 8,
        "padding": 15,
        "width": 700,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <Text
          numberOfLines={2}
          style={
            {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
          testID="Notetitle"
        />
        <Text
          numberOfLines={3}
          style={
            {
              "paddingTop": 8,
            }
          }
          testID="Notedesc"
        />
      </View>
      <View
        style={
          {
            "width": 50,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            alt="Edit"
            source={
              {
                "uri": "https://cdn-icons-png.flaticon.com/128/2951/2951136.png",
              }
            }
            style={
              {
                "height": 25,
                "marginBottom": 15,
                "marginLeft": 3,
                "width": 25,
              }
            }
            testID="Edit1"
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            alt="Delete"
            source={
              {
                "uri": "https://cdn-icons-png.flaticon.com/128/10374/10374182.png",
              }
            }
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
            testID="Delete1"
          />
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  />
</View>
`;
